src/runtime.rs

```diff
@@ -64,3 +65,15 @@ impl EthereumRelayerGame for DarwiniaRuntime {
     );
 }
 impl EthereumBacking for DarwiniaRuntime {}
+
+#[derive(Copy, Clone, Eq, PartialEq, Ord, PartialOrd, Encode, Decode)]
+pub enum ProxyType {
+    Any,
+    NonTransfer,
+    Staking,
+    IdentityJudgement,
+    EthereumBridge,
+}
```

Comment:
This `ProxyType` should be placed in `src/chain/proxy.rs`, modules under `frame` should only contain:

+ The definition of the module `trait`
+ The instance of storage(`Store`)
+ The instance of call(`Call`)

Reply:
why? the code is from chain runtime, where the `ProxyType` is inline defined.

Reply:
For example:

https://github.com/darwinia-network/bridger/blob/8f584b5953b3febc93dc47d149796bc58f4b94b9/src/api/darwinia.rs#L4-L19

```rust
use primitives::{
    chain::ethereum::{EthereumReceiptProofThing, EthereumRelayHeaderParcel, RedeemFor},
    frame::{
        collective::{ExecuteCallExt, MembersStoreExt},
        ethereum::{
            backing::{RedeemCallExt, VerifiedProofStoreExt},
            game::{AffirmationsStoreExt, EthereumRelayerGame, PendingRelayHeaderParcelsStoreExt},
            relay::{
                AffirmCallExt, ApprovePendingRelayHeaderParcel, ConfirmedBlockNumbersStoreExt,
                RejectPendingRelayHeaderParcel, SetConfirmedParcel,
            },
        },
        sudo::{KeyStoreExt, SudoCallExt},
    },
    runtime::DarwiniaRuntime,
};
```

Developers should be able to use the types of the runtime module even they didn't import the `runtime` feature to their projects —— they can only import the types to do scale codec things.

Lots of types like `EthereumReceiptProofThing, EthereumRelayHeaderParcel, RedeemFor` are used by the runtime module as well, and all of them all placed in `/src/chain/**/*`

Reply:
@freehere107 I agree with wuminzhe that this belongs to runtime, remember that we could have different runtimes, and different runtimes can have different ProxyTypes.

ProxyType is defined in Runtiime, not frame

Reply:
> ProxyType is defined in Runtime, not frame

@HackFisher  

If so, the `ProxyType` should be placed in `/src/runtime/crab/proxy.rs`, not `src/frame/proxy.rs`